src/lib/unlang/xlat_builtin.c

```diff
@@ -1117,97 +1117,98 @@ static xlat_action_t xlat_func_explode(TALLOC_CTX *ctx, fr_dcursor_t *out,
 	return XLAT_ACTION_DONE;
 }
 
+static xlat_arg_parser_t const xlat_func_integer_args[] = {
+	{ .required = true, .single = true, .type = FR_TYPE_VOID },
+	XLAT_ARG_PARSER_TERMINATOR
+};
+
 /** Print data as integer, not as VALUE.
  *
  * Example:
 @verbatim
 update request {
 	&Tmp-IP-Address-0 := "127.0.0.5"
 }
-"%{integer:&Tmp-IP-Address-0}" == 2130706437
+"%(integer:%{Tmp-IP-Address-0})" == 2130706437
 @endverbatim
  * @ingroup xlat_functions
  */
-static ssize_t xlat_func_integer(UNUSED TALLOC_CTX *ctx, char **out, size_t outlen,
-				 UNUSED void const *mod_inst, UNUSED void const *xlat_inst,
-				 request_t *request, char const *fmt)
+static xlat_action_t xlat_func_integer(TALLOC_CTX *ctx, fr_dcursor_t *out,
+				       request_t *request, UNUSED void const *xlat_inst,
+				       UNUSED void *xlat_thread_inst, fr_value_box_list_t *in)
 {
-	fr_pair_t	*vp;
+	fr_value_box_t	*in_vb = fr_dlist_head(in);
+	fr_value_box_t	*vb;
 
 	uint64_t	int64 = 0;	/* Needs to be initialised to zero */
-	uint32_t	int32 = 0;	/* Needs to be initialised to zero */
-
-	fr_skip_whitespace(fmt);
+	char		buff[40];	/* Temporary string representation special conversions */
 
-	if ((xlat_fmt_get_vp(&vp, request, fmt) < 0) || !vp) return 0;
+	switch (in_vb->type) {
+	case FR_TYPE_UINT8:
+	case FR_TYPE_UINT16:
+	case FR_TYPE_UINT32:
+	case FR_TYPE_UINT64:
+	case FR_TYPE_INT8:
+	case FR_TYPE_INT16:
+	case FR_TYPE_INT32:
+	case FR_TYPE_INT64:
+	case FR_TYPE_TIME_DELTA:
+	case FR_TYPE_SIZE:
+	case FR_TYPE_BOOL:
+		/*
+		 *	Ensure enumeration is NULL so that the integer
+		 *	version of a box is returned
+		 */
+		fr_value_box_cast_in_place(ctx, in_vb, in_vb->type, NULL);
```

Comment:
You just need to set the field in the box to NULL.  If there's issues manipulating the boxes directly add a helper function to clear the enum field in value.c